flink-runtime/src/test/java/org/apache/flink/runtime/deployment/ShuffleDescriptorTest.java

```diff
@@ -0,0 +1,187 @@
+			ResultPartitionID localPartitionId = new ResultPartitionID();
+			ResultPartitionDeploymentDescriptor localPartition =
+				createResultPartitionDeploymentDescriptor(localPartitionId, consumerResourceID);
+
+			ResultPartitionID remotePartitionId = new ResultPartitionID();
+			ResultPartitionDeploymentDescriptor remotePartition =
+				createResultPartitionDeploymentDescriptor(remotePartitionId, ResourceID.generate());
+
+			ResultPartitionID unknownPartitionId = new ResultPartitionID();
+
+			ShuffleDescriptor localShuffleDescriptor =
+				getConsumedPartitionShuffleDescriptor(localPartitionId, state, localPartition, true);
+			ShuffleDescriptor remoteShuffleDescriptor =
+				getConsumedPartitionShuffleDescriptor(remotePartitionId, state, remotePartition, true);
+			ShuffleDescriptor unknownShuffleDescriptor =
+				getConsumedPartitionShuffleDescriptor(unknownPartitionId, state, null, true);
+
+			// These states are allowed
+			if (state == ExecutionState.RUNNING ||
+				state == ExecutionState.FINISHED ||
+				state == ExecutionState.SCHEDULED ||
+				state == ExecutionState.DEPLOYING) {
+				NettyShuffleDescriptor nettyShuffleDescriptor;
+
+				// Create local or remote channels
+				assertThat(localShuffleDescriptor, instanceOf(NettyShuffleDescriptor.class));
+				assertThat(localShuffleDescriptor.isUnknown(), is(false));
+				nettyShuffleDescriptor = (NettyShuffleDescriptor) localShuffleDescriptor;
+				assertThat(nettyShuffleDescriptor.getResultPartitionID(), is(localPartitionId));
+				assertThat(nettyShuffleDescriptor.isLocalTo(consumerResourceID), is(true));
+
+				assertThat(remoteShuffleDescriptor, instanceOf(NettyShuffleDescriptor.class));
+				assertThat(remoteShuffleDescriptor.isUnknown(), is(false));
+				nettyShuffleDescriptor = (NettyShuffleDescriptor) remoteShuffleDescriptor;
+				assertThat(nettyShuffleDescriptor.getResultPartitionID(), is(remotePartitionId));
+				assertThat(nettyShuffleDescriptor.isLocalTo(consumerResourceID), is(false));
+				assertThat(nettyShuffleDescriptor.getConnectionId(), is(CONNECTION));
+			} else {
+				// Unknown (lazy deployment allowed)
+				assertThat(localShuffleDescriptor, instanceOf(UnknownShuffleDescriptor.class));
+				assertThat(localShuffleDescriptor.isUnknown(), is(true));
+				assertThat(localShuffleDescriptor.getResultPartitionID(), is(localPartitionId));
+
+				assertThat(remoteShuffleDescriptor, instanceOf(UnknownShuffleDescriptor.class));
+				assertThat(remoteShuffleDescriptor.isUnknown(), is(true));
+				assertThat(remoteShuffleDescriptor.getResultPartitionID(), is(remotePartitionId));
+			}
+
+			assertThat(unknownShuffleDescriptor, instanceOf(UnknownShuffleDescriptor.class));
+			assertThat(unknownShuffleDescriptor.isUnknown(), is(true));
+			assertThat(unknownShuffleDescriptor.getResultPartitionID(), is(unknownPartitionId));
+		}
+	}
+
+	@Test
+	public void testUnknownChannelWithoutLazyDeploymentThrows() throws Exception {
```

Comment:
`testUnknownChannelWithoutLazyDeploymentThrows` seems not easy to understand, maybe called `testUnknownDescriptorWithOrWithoutLazyDeployment`?